### PR TITLE
Use linz-software-repository github action to build packages on every commit

### PR DESCRIPTION
**Pull request by strk**
Very early step here, mostly trying out the github action
workflow...

Ref: https://github.com/linz/linz-software-repository/issues/19

### CODE REVIEW

**Comment by strk** [comment]
Small problem: it looks like github is caching the Dockerfile of an action, so changes I already pushed to the action repository are not visible on next build triggered by pushing commits here (in this case, I changed Dockerfile of linz-software-repository to have a default DISTRIBUTION of `bionic` while new commit here is still seeing `latest` as the default)

**Comment by strk** [comment]
Success! Packages build, now on to decide when to build and when to publish

**Comment by strk** [comment]
When https://github.com/linz/linz-software-repository/pull/21 is merged, and `v1` is tagged, we should change the action name specification to use `@v1` instead of `@github-action` as it is now

**Comment by strk** [comment]
One issue with current setup is that the workflow builds debian packages once for _each_ build matrix cell. There's no need for this because packages are built using the docker image, so the test won't change result based on matrix. We could save cycles by testing debian packaging only once...

**Comment by strk** [comment]
@imincik: we need an API token from packagecloud in the secrets, but I don't see the ability of creating multiple tokens, so right now I could only give my own API token. Shall we create an ad-hoc account on packagecloud for this kind of action ? Like a linz-github-action account...

**Comment by strk** [comment]
Current problem is the naming of packages. It looks like the package name remains 1.5.0 even if Makefile expresses a version of `1.6.0dev` I think this is due to the packager using latest tag instead of current version, which is good for a tagged release but not so good for intermediate packages (shall we decide to push packages on every commit, or just for testing purposes)

**Comment by strk** [comment]
Fun fact:
```
# dch: fatal error at line 1092:
New version specified (1.5.0-2linz~bionic1) is less than
the current version number (1.5.0-3linz~bionic1)!  Use -b to force.
```

But if the versions are _the_same_ there is no fatal error, simply you end up with multiple debian/changelog sections for the _same_ version. How bad is that ? I would report upstream, can you help me find out where's the upstream of `dch` @imincik  ?

**Comment by imincik** [comment]
`dch` is part of `devscripts`. Here is Debian way how to report bugs - https://www.debian.org/Bugs/Reporting#reportbug

**Comment by strk** [comment]
@imincik ok I was about to send a bug but then I realized the behaviour was different when using the `-D` switch and not using it. For now I postponed that part. Instead I made changes to linz-software-repository so that package version is extracted from `git describe` which gives a nice serial based on number of commits since last tag. I like [what I see](https://github.com/linz/postgresql-dbpatch/pull/67/checks?check_run_id=692974606#step:6:4) as version numbers now, do you like them ? 

The change in linz-software-repository is in https://github.com/linz/linz-software-repository/pull/23 - can you please review that ?

**Comment by strk** [comment]
I think upload is failing because secrets are not available to pull requests from forks.
It makes sense anyway to only upload packages upon merging a PR, not while testing it, so will change the code accordingly. Of course there'll be no way to tell if it works until we merge, ok to proceed like this @imincik ?

Also, I think we could use a packagecloud action, like https://github.com/RadicalZephyr/packagecloud/blob/master/action.yml

My understanding is that the docker image will be built by ourselves, so unless the action itself communicates secrets elsewhere we would be "safe" . Of course referencing the action by TAG or BRANCH would make it impossible for us to tell if the author will turn that action into an insecure one, but referencing it by commit-hash should be safe.

**Comment by strk** [comment]
@imincik this is ready for review now. It only publishes packages when either tagging or pushing to "master" or "packagecloud" branches.

It tests packages in every pull-request + before publishing.

Trigger on tag is yet to be tested (as it needs a tag to test)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
 name: test
 
-on: [pull_request]
+on:
+  pull_request:
```
Comment (imincik) [inline .github/workflows/test.yml:4]: @strk , is this condition really correct ? Are we going to run (and package and package upload) for each pull request ? We want to upload packages to `dev` only from `master` branch.

Comment (strk) [reply .github/workflows/test.yml:4]: We have a single workflow here, so it runs BOTH on push and on pull-request.
One step in the workflow (under the "packaging" step) is to DETERMINE publication target.
Possible targets are: 

  - '' (used on pull-request, meaining NO publication)
  - test (used on tag)
  - dev (used on push)

Comment (strk) [reply .github/workflows/test.yml:4]: See for example https://github.com/linz/linz-lds-bde-schema/runs/708783919?check_suite_focus=true#step:4:19
The run above is for a pull-request, REPO is empty (means no publication)

Comment (imincik) [reply .github/workflows/test.yml:4]: Thanks for clarification.

Comment (strk) [reply .github/workflows/test.yml:4]: you need to approve this, or I cannot merge

**Comment by strk** [comment]
nevermind, I used admin powers

**Pull request merged**